### PR TITLE
JBPM-7504 Not possible to edit existing Deployment Descriptor configu…

### PR DESCRIPTION
**Pull request by bxf12315**
…ration, for inline edit

@cristianonicolai Please review it.

### CODE REVIEW

**Comment by bxf12315** [comment]
@cristianonicolai According Matthew's suggestion, I change add new workitem way to inline add.
and remove pencil icon. 
Please review it.

**Comment by bxf12315** [comment]
@cristianonicolai 
I was refactor all Section config of jbpm-wb and working. I find a unit of org.jbpm.workbench.wi.client.handlers.NewCaseProjectHandlerTest. is not work.
Error in here  
in org.jbpm.workbench.wi.client.handlers.NewCaseProjectHandlerTest
testNewCaseProject(org.jbpm.workbench.wi.client.handlers.NewCaseProjectHandlerTest)  Time elapsed: 0.062 sec  <<< ERROR!
java.lang.ArrayIndexOutOfBoundsException: 0
	at org.uberfire.mocks.MockInstanceImpl.get(MockInstanceImpl.java:67)
	at org.jbpm.workbench.wi.client.handlers.NewCaseProjectHandlerTest.testNewCaseProject(NewCaseProjectHandlerTest.java:74)
The test not belong to the PR. and I can not find the reason.
Did I miss some thing ?
Thanks

**Comment by bxf12315** [comment]
@cristianonicolai  I was fix the test error issue . Please review it.

**Comment by cristianonicolai** [comment]
@bxf12315 this error is fixed already on master, please rebase your PR.

File: jbpm-wb-integration/jbpm-wb-integration-client/src/main/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/NamedObjectItemPresenter.java [hunk 1]
```diff
@@ -111,5 +131,9 @@ public void signalParameterAddedOrRemoved() {
         void setValue(final String value);
 
         void setParametersCount(final int size);
+
+        HTMLInputElement getNameElement();
```
Comment (cristianonicolai) [inline jbpm-wb-integration/jbpm-wb-integration-client/src/main/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/NamedObjectItemPresenter.java]: Please avoid manipulating view elements in presenter logic.

Comment (bxf12315) [reply jbpm-wb-integration/jbpm-wb-integration-client/src/main/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/NamedObjectItemPresenter.java]: yes you are right, I updated it .Please review again.

File: jbpm-wb-integration/jbpm-wb-integration-client/src/test/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/ObjectItemPresenterTest.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 package org.jbpm.workbench.wi.client.editors.deployment.descriptor.items;
 
 import com.google.common.collect.ImmutableList;
+import elemental2.dom.HTMLInputElement;
```
Comment (cristianonicolai) [inline jbpm-wb-integration/jbpm-wb-integration-client/src/test/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/ObjectItemPresenterTest.java]: Changes not needed?

File: jbpm-wb-integration/jbpm-wb-integration-api/src/main/java/org/jbpm/workbench/wi/dd/model/SingleValueItemObjectModel.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.wi.dd.model;
+
+import org.jboss.errai.common.client.api.annotations.Portable;
+
+@Portable
+public class SingleValueItemObjectModel {
```
Comment (cristianonicolai) [inline jbpm-wb-integration/jbpm-wb-integration-api/src/main/java/org/jbpm/workbench/wi/dd/model/SingleValueItemObjectModel.java]: please reuse class from kie-wb-common

File: jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/dd/DDEditorServiceImpl.java [hunk 4]
```diff
@@ -315,10 +317,18 @@ protected DeploymentDescriptorModel marshal(DeploymentDescriptor originDD) {
         ddModel.setConfiguration(processNamedObjectModel(configuration));
 
         // required roles
-        ddModel.setRequiredRoles(originDD.getRequiredRoles());
+        List<SingleValueItemObjectModel> roles = new ArrayList<SingleValueItemObjectModel>();
```
Comment (cristianonicolai) [inline jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/dd/DDEditorServiceImpl.java]: same as comment regarding functional style.

File: jbpm-wb-integration/jbpm-wb-integration-client/src/main/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/NamedObjectItemPresenter.java [hunk 5]
```diff
@@ -110,6 +128,12 @@ public void signalParameterAddedOrRemoved() {
 
         void setValue(final String value);
 
+        String getName();
```
Comment (cristianonicolai) [inline jbpm-wb-integration/jbpm-wb-integration-client/src/main/java/org/jbpm/workbench/wi/client/editors/deployment/descriptor/items/NamedObjectItemPresenter.java]: I think you will find that these gettter methods arent needed once you can the presenter passing the new value.

**Comment by bxf12315** [comment]
@cristianonicolai as your comment, I updated the PR.Please review it again.

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: jbpm-wb-integration/jbpm-wb-integration-api/src/main/java/org/jbpm/workbench/wi/dd/model/SingleValueItemObjectModel.java (hunk 3 above)
Comment (cristianonicolai) [reply jbpm-wb-integration/jbpm-wb-integration-api/src/main/java/org/jbpm/workbench/wi/dd/model/SingleValueItemObjectModel.java]: should it be removed?

**Comment by bxf12315** [comment]
@cristianonicolai  updated, Please review it

**Comment by cristianonicolai** [comment]
@bxf12315 the branch name needs to match with https://github.com/kiegroup/kie-wb-common/pull/2132 for the compilation to work.

**Comment by cristianonicolai** [comment]
Replaced by https://github.com/kiegroup/jbpm-wb/pull/1213

**Pull request closed without merging**